pkg/controller/accountclaim/organizational_units.go

```diff
@@ -52,13 +52,17 @@ func MoveAccountToOU(r *ReconcileAccountClaim, reqLogger logr.Logger, accountCla
 	err = MoveAccount(reqLogger, awsClient, account, OUID, rootID)
 	if err != nil {
 		// If error was cause by the account already being inside the OU, simply update the accountclaim cr and returns
-		if err == awsv1alpha1.ErrAccAlreadyInOU {
+		switch err {
+		case awsv1alpha1.ErrAccAlreadyInOU:
 			// Log account already in desired location
 			accountMovedMsg := fmt.Sprintf("OU: Account %s was already in the desired OU %s", account.Name, account.Spec.LegalEntity.ID)
 			reqLogger.Info(accountMovedMsg)
 			// Update accountclaim spec
 			accountClaim.Spec.AccountOU = OUID
 			return r.specUpdate(reqLogger, accountClaim)
+		case awsv1alpha1.ErrAccMoveRaceCondition:
+			// Simply return the error since we want a requeue
+			return awsv1alpha1.ErrAccMoveRaceCondition
```

Comment:
Sorry, what I meant here is that this whole thing could have been left as is. The race condition case is the same as the fall-through, simply returning the `err`. So the only special thing that needs to be done here is if `ErrAccAlreadiInOU`, which is how it was written before.

Not a blocking issue, as the logic is (still) correct. Just an opportunity to clean up a little bit :)

Reply:
You are right, I'll update it once again